### PR TITLE
Smallpatches

### PR DESCRIPTION
**Pull request by InfusOnWoW**

### CODE REVIEW

File: WeakAurasOptions/BuffTrigger2.lua [hunk 1]
```diff
@@ -578,6 +578,62 @@ local function GetBuffTriggerOptions(data, triggernum)
...
+    },
+    use_castByPlayer = {
+      type = "toggle",
+      name = function(input)
+        local value = trigger.use_castByPlayer
+        if value == nil then return L["Cast by Player Character"]
+        elseif value == false then return "|cFFFF0000 "..L["Negator"].." "..L["Cast by Player Character"]
+        else return "|cFF00FF00"..L["Cast by Player Character"] end
+      end,
+      width = WeakAuras.doubleWidth,
```
Comment (mrbuds) [inline WeakAurasOptions/BuffTrigger2.lua:617]:
```Lua
desc = L["Only Match auras cast by a player (not an npc)"],
```

File: WeakAuras/Prototypes.lua [hunk 2]
```diff
@@ -1607,9 +1607,7 @@ Private.event_prototypes = {
       AddUnitEventForEvents(result, unit, "UNIT_LEVEL")
       AddUnitEventForEvents(result, unit, "UNIT_FACTION")
       AddUnitEventForEvents(result, unit, "UNIT_NAME_UPDATE")
-      if trigger.use_ignoreDead or trigger.use_ignoreDisconnected then
-        AddUnitEventForEvents(result, unit, "UNIT_FLAGS")
-      end
+      AddUnitEventForEvents(result, unit, "UNIT_FLAGS")
```
Comment (mrbuds) [inline WeakAuras/Prototypes.lua:1610]: on classic with unit = nameplate losing aggro doesn't send UNIT_FLAGS on nameplate unit, but send a UNIT_THREAT_LIST_UPDATE with arg1 = "nameplate1"

Comment (mrbuds) [reply WeakAuras/Prototypes.lua:1610]: by "losing aggro" i mean walking away with nameplate visible until i'm out of combat and npc go away

Comment (InfusOnWoW) [reply WeakAuras/Prototypes.lua:1610]: Hmm, that probably fires a bit too often. Any other events?

Comment (mrbuds) [reply WeakAuras/Prototypes.lua:1610]: The only other event that trigger for "nameplate1" (i don't target it) is UNIT_MAXHEALTH

Comment (InfusOnWoW) [reply WeakAuras/Prototypes.lua:1610]: Hmm, okay. I'll check what the behaviour on retail is.

Comment (mrbuds) [reply WeakAuras/Prototypes.lua:1610]: After more tests, It seems UNIT_FLAGS correctly trigger when UnitAffectingCombat change from true to false after a npc has reset aggro by going away.
I think it's a blizzard (non-)issue that the combat state change a few seconds after npc has fully reset (it's very slow)

I was testing with 
```
!WA:2!DjvWUnTsuuIksG8cenuvbBqvVfiQenkVwAQELwHStDOHNtJXXfyrKsh75A7r1EgRzghAlsSiRyD(eY63Q8j0fi1Dwv8fW23U(f8MXPVUaVXhDM5EUN75o16upRoUo(7RntYchdCbHrF4Rw)gjCMSGdFIJY7XWq(kTDm75633YXS9FBEWbD97(r7RrfYegVFUuvLWW7leSm5k(z9JIeGSMxilLXF)sh(2)9R6VVvZ9(nBSZ2bzeoNXt4NV4I7phrdv64Yiu5M(9D9qP5jOLMgKcuSU9Ewo2hDWmoeRAK)55a)w)zLYq4wwICiCvVSIujXWlmfjeRw(edlb5cqdMfYOyYctYrHvGwEcjIlncIiuIiXWs9tAu(4896sLapcfcdnX4(uXWpbOtnl4iXWEaMGg6VO3IHkbIbjHgVrgIFAdzmAIKtIJvX4JEb)w4pSKkdhO6MSqWlOWyGkTujhBcwjQ2m1)tVk28vowzI1ANG0Ue4eHKekMuiGrfQdo5snIqBZYcqYtMOM600UyHHL(0PuugKNIKGNgjm(LOiOswxoeroZBGRTJZDKdkIuKZh12CG)Ob(ME(ZWqGISkEp02XTZXotlO3ofRw(YY1ZxwhEJb)fC6nZ)CVBi68IIs)4IxpFiFzmreYbjmIZKvdyTPcinQA)g02(iFBV)GGNsORfwnktrusw1fBv(0oxcibmqYbASm5bLp3IYOW1)FwnsNMCbOxPIj6RQnC5UwzicTt5FPkOC3Y3uUx5(Q)p73zMhXvHJsDKeTKAIfQTIcdjtUZRbkPJiXg3S4vzhDb6Em9iZE2UoM(2ZuHOx1BrHrqcqItKxfuvpCYCcnIXxmngBwqWZB6L)UTO9rBFrCDr(YB1OzJwBSzZnIB(6DcWaU(4F(5)7
```
and an every frame check
```
!WA:2!1nzZUTTrqCClOdoGiOnwOiafThyvrkIbICDClCApkkqHMufjhk660Erzj3zL2aYDj2pST8rIEON9Jap3t6rOh6zcJ(e4hH8eKzjvsdcA5foZWzNp(T)zNX9Y7r7r)JhOKwtgxaQzNgp5PtdJsLzs1Z6IpBOCDrgzDmCPP)dsRsTAJm35vmLzfPgUu8W998XhfySkH)PcUziJb4NelhjZtiMh23qulbt)99jcQFFJYc99Lk)(msMgn9ab9g9kcvEXVoJX0G5l3zdrKUsQorYfMKrHtJdJkU3)2(tlOedKyBEDlXIHjgE6zCQzvWq0vrAgo9XrAS7gVaooyEjm8LELxPrXxUeu6p5BuBn)7aZ6cibZ2y13A1Wcs2fK1A30(QB12e4CqyMBzm(LBwmA484fZJhgfxsTkIRv9ECutkvJKckVP5LUYyXg)QsDbKL9uQ279L6efGLkA(jHtMOSIMybUnjsqYbTxG7Gjo6dQkkURmwmoHQFkCYjJpDY1wX2j3Ryp3YEoe36)Cjf(ZDk(Ip4UBbtQqaTOyRrKHNd03GSfucs2VGKaN3xCnrWZBwMJR)0X1FwGqkG)ciAyUrbILMv7EZ7w3foCP0qkUT6sxoUPREqqoHlWZwpO(GDR)26dRFmAU3hhjGjfMI7nwXVY)fwcflkXpoUcPteS0bpVGmjHECasU07hLBZmCVO0mIwF)6FWlqZVcCgLUcnhD(8TsOx2kH6wPAQJBQcmidQqAd0gfYVF3IV()KooSSOqbPChp6w8v))zrxJxt80vxRHmwRmnywC8SNREnQs5S1btchhxzKPN3s378O97ZPhnfU4ilNUH)ZYztep73O)4WnmfEJJigzq330k8h7c5M9K5JIcdNQw3UwDsqGZ4l9uxUnWnOMbpWSIgjN3MwkmQ5F4oD60Tk996rVnCr7wGEELxiv0ZuKIYZ2A0txS33DWHhC8GJoCWYd)(NKqbAVZ)Nx(2d
```
And nameplate update happen when the function called every frame change it's returned value

File: WeakAurasOptions/BuffTrigger2.lua (hunk 1 above)
Comment (Nightwarden24) [reply WeakAurasOptions/BuffTrigger2.lua:617]: Yea, description will definitely be helpful here. I would like to see its implementation as in the case of "Own only" (two descriptions; a couple of lines below ↓)

**Pull request merged**